src/test/resources/examples/discriminatedOneOf/models/Models.kt

```diff
@@ -0,0 +1,53 @@
+package examples.discriminatedOneOf.models
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonValue
+import javax.validation.Valid
+import javax.validation.constraints.NotNull
+import kotlin.String
+import kotlin.collections.Map
+
+data class SomeObj(
+    @param:JsonProperty("state")
+    @get:JsonProperty("state")
+    @get:NotNull
+    @get:Valid
+    val state: State
+)
+
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.EXISTING_PROPERTY,
+    property = "status",
+    visible = true
+)
+@JsonSubTypes(
+    JsonSubTypes.Type(value = StateA::class, name = "a"),
+    JsonSubTypes.Type(
+        value =
+        StateB::class,
+        name = "b"
+    )
+)
+sealed interface State
```

Comment:
How does Jackson handle this? We're instructing it that a status property is the discriminator, but none of the sub-types contain the status property

Reply:
Jackson doesn't care if the property is part of the class. It decodes it just fine

Reply:
Ah cool, understood.